src/net.cpp

```diff
@@ -2539,15 +2540,47 @@ std::vector<CAddress> CConnman::GetAddresses(size_t max_addresses, size_t max_pc
     return addresses;
 }
 
-std::vector<CAddress> CConnman::GetAddresses(Network requestor_network, size_t max_addresses, size_t max_pct)
+std::vector<CAddress> CConnman::GetAddresses(CNode& requestor, size_t max_addresses, size_t max_pct)
 {
+    SOCKET socket;
+    WITH_LOCK(requestor.cs_hSocket, socket = requestor.hSocket);
+    auto local_socket_bytes = GetBindAddress(socket).GetAddrBytes();
```

Comment:
Can this be simplified like:

```suggestion
    auto local_socket_bytes = requestor.addrBind.GetAddrBytes();
```

Our local address is already saved in `CNode::addrBind`.

Reply:
I think you're right?

Reply:
Ok, will include this tiny change in another PR:

<details>
<summary>net: avoid unnecessary GetBindAddress() call</summary>

```diff
commit 15ba60e9a
Parent: 25dcd4ad9
Author:     Vasil Dimov <vd@FreeBSD.org>
AuthorDate: Tue Nov 24 15:36:27 2020 +0100
Commit:     Vasil Dimov <vd@FreeBSD.org>
CommitDate: Tue Nov 24 15:36:27 2020 +0100

    net: avoid unnecessary GetBindAddress() call
    
    Our local (bind) address is already saved in `CNode::addrBind` and there
    is no need to re-retrieve it again with `GetBindAddress()`.
    
    Also, for I2P connections `CNode::addrBind` would contain our I2P
    address, but `GetBindAddress()` would return something like
    `127.0.0.1:RANDOM_PORT`.

diff --git a/src/net.cpp b/src/net.cpp
index a811732e2..94a424ca3 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -2601,15 +2601,13 @@ std::vector<CAddress> CConnman::GetAddresses(size_t max_addresses, size_t max_pc
     }
     return addresses;
 }
 
 std::vector<CAddress> CConnman::GetAddresses(CNode& requestor, size_t max_addresses, size_t max_pct)
 {
-    SOCKET socket;
-    WITH_LOCK(requestor.cs_hSocket, socket = requestor.hSocket);
-    auto local_socket_bytes = GetBindAddress(socket).GetAddrBytes();
+    auto local_socket_bytes = requestor.addrBind.GetAddrBytes();
     uint64_t cache_id = GetDeterministicRandomizer(RANDOMIZER_ID_ADDRCACHE)
         .Write(requestor.addr.GetNetwork())
         .Write(local_socket_bytes.data(), local_socket_bytes.size())
         .Finalize();
     const auto current_time = GetTime<std::chrono::microseconds>();
     auto r = m_addr_response_caches.emplace(cache_id, CachedAddrResponse{});
```
</details>

You just reviewed that! :partying_face: